packages/opentelemetry-core/test/correlation-context/HttpCorrelationContext.test.ts

```diff
@@ -0,0 +1,172 @@
+  setCorrelationContext,
+} from '../../src/correlation-context/correlation-context';
+
+import {
+  HttpCorrelationContext,
+  CORRELATION_CONTEXT_HEADER,
+  MAX_PER_NAME_VALUE_PAIRS,
+} from '../../src/correlation-context/propagation/HttpCorrelationContext';
+
+describe('HttpCorrelationContext', () => {
+  const httpTraceContext = new HttpCorrelationContext();
+
+  let carrier: { [key: string]: unknown };
+
+  beforeEach(() => {
+    carrier = {};
+  });
+
+  describe('.inject()', () => {
+    it('should set traceparent header', () => {
+      const correlationContext: CorrelationContext = {
+        key1: { value: 'd4cda95b652f4a1592b449d5929fda1b' },
+        key3: { value: 'c88815a7-0fa9-4d95-a1f1-cdccce3c5c2a' },
+        'with/slash': { value: 'with spaces' },
+      };
+
+      httpTraceContext.inject(
+        setCorrelationContext(Context.ROOT_CONTEXT, correlationContext),
+        carrier,
+        defaultSetter
+      );
+      assert.deepStrictEqual(
+        carrier[CORRELATION_CONTEXT_HEADER],
+        'key1=d4cda95b652f4a1592b449d5929fda1b,key3=c88815a7-0fa9-4d95-a1f1-cdccce3c5c2a,with%2Fslash=with%20spaces'
+      );
+    });
+
+    it('should skip long key-value pairs', () => {
+      const correlationContext: CorrelationContext = {
+        key1: { value: 'd4cda95b' },
+        key3: { value: 'c88815a7' },
+      };
+
+      let value = '';
+      // Generate long value 2*MAX_PER_NAME_VALUE_PAIRS
+      for (let i = 0; i < MAX_PER_NAME_VALUE_PAIRS; ++i) {
+        value += '1a';
+      }
+      correlationContext['longPair'] = { value };
+
+      httpTraceContext.inject(
+        setCorrelationContext(Context.ROOT_CONTEXT, correlationContext),
+        carrier,
+        defaultSetter
+      );
+      assert.deepStrictEqual(
+        carrier[CORRELATION_CONTEXT_HEADER],
+        'key1=d4cda95b,key3=c88815a7'
+      );
+    });
+
+    it('should skip all keys that surpassed the max limit of the header', () => {
+      const correlationContext: CorrelationContext = {};
+
+      const zeroPad = (num: number, places: number) =>
+        String(num).padStart(places, '0');
+
+      // key=value with same size , 1024 => 8 keys
+      for (let i = 0; i < 9; ++i) {
+        const index = zeroPad(i, 510);
+        correlationContext[`k${index}`] = { value: `${index}` };
+      }
+
+      // Build expected
+      let expected = '';
+      for (let i = 0; i < 8; ++i) {
+        const index = zeroPad(i, 510);
+        expected += `k${index}=${index},`;
+      }
+      expected = expected.slice(0, -1);
+
+      httpTraceContext.inject(
+        setCorrelationContext(Context.ROOT_CONTEXT, correlationContext),
+        carrier,
+        defaultSetter
+      );
+      assert.deepStrictEqual(carrier[CORRELATION_CONTEXT_HEADER], expected);
+    });
+  });
+
+  describe('.extract()', () => {
+    it('should extract context of a sampled span from carrier', () => {
+      carrier[CORRELATION_CONTEXT_HEADER] = 'key1=d4cda95b,key3=c88815a7';
```

Comment:
Could you please add tests to valid header with multiple spaces. ex: `"key1 =   val1,  key2 =val2   "`